### PR TITLE
[oracle-cdc]Optimized slow oracle real-time synchronization

### PR DESCRIPTION
**Pull request by weAreFriendYo**
#https://github.com/ververica/flink-cdc-connectors/issues/1999
To solve the problem of slow real-time oracle synchronization, add a table name as a query condition, improving query efficiency and reducing memory usage

### CODE REVIEW

**Comment by e-mhui** [comment]
:+1: This PR can improve the speed of querying tableid, but the select operation during the full stage is still slow.

File: flink-connector-oracle-cdc/src/main/java/com/ververica/cdc/connectors/oracle/source/utils/OracleConnectionUtils.java [hunk 1]
```diff
@@ -81,17 +83,30 @@ public static RedoLogOffset currentRedoLogOffset(JdbcConnection jdbc) {
...
+        StringBuilder queryTablesSql =
+                new StringBuilder(
+                        "SELECT OWNER ,TABLE_NAME,TABLESPACE_NAME FROM ALL_TABLES \n"
+                                + "WHERE TABLESPACE_NAME IS NOT NULL AND TABLESPACE_NAME NOT IN ('SYSTEM','SYSAUX')");
+        if (tableList != null && !tableList.isEmpty()) {
+            String schema = jdbcConnection.connection().getSchema();
+            StringJoiner stringJoiner = new StringJoiner(",");
+            for (String tableId : tableList) {
+                String tableName = tableId.replaceFirst(schema + "\\.", "");
+                stringJoiner.add("'" + tableName + "'");
```
Comment (ruanhang1993) [inline flink-connector-oracle-cdc/src/main/java/com/ververica/cdc/connectors/oracle/source/utils/OracleConnectionUtils.java:100]: What if the user use regular expressions in the table names?
Maybe we need a test to cover this situation.

**Review by ruanhang1993 — COMMENTED** [review]
@weAreFriendYo Thanks for contributing. Left a comment.

**Comment by superfive666** [comment]
Any updates on this issue bug? Encountered the same slowness issue for oracle with large number of tables.

**Comment by yuxiqian** [comment]
Thanks @weAreFriendYo for the great work! Before this PR could be merged, could you please rebase it with latest `master` branch?

**Pull request closed without merging**